src/crimson/common/intrusive_timer.h

```diff
@@ -0,0 +1,136 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <boost/intrusive/set.hpp>
+#include <boost/intrusive/list.hpp>
+
+#include <seastar/core/future.hh>
+#include <seastar/core/condition-variable.hh>
+
+#include "include/ceph_assert.h"
+#include "common/ceph_time.h"
+#include "crimson/common/log.h"
+
+namespace crimson::common {
+
+/**
+ * intrusive_timer: timer implementation with allocation-free
+ * schedule/cancel
+ */
+class intrusive_timer_t {
+  using clock_t = ceph::coarse_real_clock;
```

Comment:
Is it better to use seastar native `seastar::lowres_system_clock` ?

Reply:
Nope, that works perfectly -- I just missed it.  I've repushed with seastar::timer and am running another test set.